### PR TITLE
Additional TabController, DefaultTabController docs

### LINKED ISSUE #7926: DefaultTabController should explain how it relates to TabController
DefaultTabController and TabController have similar names but it is unclear how they relate to each other.  We should explain in more detail.

/cc @HansMuller

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes https://github.com/flutter/flutter/issues/7926

### CODE REVIEW

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 1]
```diff
@@ -13,10 +13,47 @@ import 'constants.dart';
...
+///     new Tab(text: 'LEFT'),
+///     new Tab(text: 'RIGHT'),
+///   ];
+///
+///   TabController _tabController;
+///
+///   @override
+///   void initState() {
+///     super.initState();
+///     _tabController = new TabController(vsync: this, length: myTabs.length);
```
Comment (abarth) [inline packages/flutter/lib/src/material/tab_controller.dart]: Do you need to dispose the tab controller in `dispose` ?

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/tab_controller.dart (hunk 1 above)
Comment (HansMuller) [inline packages/flutter/lib/src/material/tab_controller.dart]: Yes, thanks for catching that!

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 2]
```diff
@@ -13,10 +13,53 @@ import 'constants.dart';
 /// represents the current scroll positions of the tab bar and the tar bar view.
 /// The selected tab's index can be changed with [animateTo].
 ///
-/// See also:
+/// A stateful widget that builds a [TabBar] and/or a [TabBarView] can create
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tab_controller.dart]: nit: I would avoid "and/or". Just say "or".

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 3]
```diff
@@ -13,10 +13,53 @@ import 'constants.dart';
...
+/// create a TabController and share it directly.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tab_controller.dart]: nit: create create

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 4]
```diff
@@ -139,6 +182,40 @@ class _TabControllerScope extends InheritedWidget {
 }
 
 /// The [TabController] for descendant widgets that don't specify one explicitly.
+///
+/// DefaultTabController is an inherited widget that's used to share a
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tab_controller.dart]: that's -> that is

**Pull request merged**